[Fix #6861] Fix a false positive for `Layout/IndentationWidth`

koic opened the pull request:
Fixes #6861.

This is a regression by https://github.com/rubocop-hq/rubocop/pull/6792.

This PR fixes infinite loop for `Layout/IndentationWidth` cop and `Layout/AccessModifierIndentation` cop.

```ruby
# exmaple.rb
class Foo
private

  def do_something
  end
end
```

```yaml
# .rubocop.yml
% cat .rubocop.yml
Layout/AccessModifierIndentation:
  EnforcedStyle: outdent
```

First, auto-corrected by `Layout/IndentationWidth`.

```console
% rubocop -v
0.66.0
% rubocop -a --only Layout/IndentationWidth
Inspecting 1 file
C

Offenses:

example.rb:2:1: C: [Corrected] Layout/IndentationWidth: Use 2 (not 0)
spaces for indentation.
private

1 file inspected, 1 offense detected, 1 offense corrected
```

```diff
% git diff example.rb
diff --git a/example.rb b/example.rb
index 063e6a5..081b80a 100644
--- a/example.rb
+++ b/example.rb
@@ -1,5 +1,5 @@
 class Foo
-private
+  private

   def do_something
   end
```

Next, auto-corrected by `Layout/Layout/AccessModifierIndentation` (`EnforcedStyle: outdent`).

```console
% rubocop -a --only Layout/AccessModifierIndentation
Inspecting 1 file
C

Offenses:

example.rb:2:3: C: [Corrected] Layout/AccessModifierIndentation: Outdent
access modifiers like private.
  private
  ^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
```

This will return to the original code.

```diff
% git diff
diff --git a/example.rb b/example.rb
index 081b80a..063e6a5 100644
--- a/example.rb
+++ b/example.rb
@@ -1,5 +1,5 @@
 class Foo
-  private
+private

   def do_something
   end
```

That caused the infinite loop in `Layout/IndentationWidth` and `Layout/AccessModifierIndentation` (`EnforcedStyle: outdent`).

With this PR, `Layout/indentationWidth` cop makes aware of `Layout/AccessModifierIndentation` cop.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

koic commented on lib/rubocop/cop/layout/indentation_width.rb:
```diff
@@ -165,18 +165,16 @@ def autocorrect(node)
-
-              check_indentation(base, member)
-            end
+            check_members_for_non_rails_style(base, members)
```
📝I extracted to the method because it was warned by `Metrics` cops.

jonas054 commented:
Hi @koic,
Great description of the problem and solution. I found one remaining problem. If I start with `EnforcedStyle: outdent` for `Layout/AccessModifierIndentation` and
```ruby
# example.rb
class Foo
                 private

  def do_something
    # something
  end
end
```
Auto-correction changes it to
```ruby
# example.rb
class Foo
private

                 def do_something
                   # something
                 end
end
```

koic commented:
@jonas054 Sure! I was not aware of this problem 💦 I'll look it up. Thanks for your checking.

deivid-rodriguez commented:
Hello @koic! Thanks for working on this, do you still plan to wrap this up?

mvastola commented:
Wanted to second merging this PR once the remaining issues are fixed. If it helps, this only seems to be a problem if the access modifier is the first line of code in the class/module. So maybe the issue with autocorrect could be fixed by limiting changes to the first line of code in this case?

bbatsov commented:
@koic What's the state here?

koic commented:
Sorry I'm late.  I updated this PR and added the following mentioned issue as a test.
https://github.com/rubocop-hq/rubocop/pull/6866#issuecomment-478321279

deivid-rodriguez commented:
Thanks for fixing this @koic!! :heart:
